internal/juju/models.go

```diff
@@ -165,12 +165,18 @@ func (c *modelsClient) CreateModel(input CreateModelInput) (*CreateModelResponse
 	}
 
 	// set constraints when required
-	if input.Constraints.String() != "" {
+	if input.Constraints.String() == "" {
 		return &CreateModelResponse{ModelInfo: modelInfo}, nil
 	}
 
-	// we have to set constraints
-	modelClient := modelconfig.NewClient(conn)
+	// we have to set constraints ...
+	// stablish a new connection with the target model and set constraints
```

Comment:
this comment seems a bit confusing

```suggestion
	// establish a new connection with the created model through the modelconfig api to set constraints
```